### DISCUSSION TITLE
Issue with adding Attributes for account Management Console to custum profile

### DISCUSSION THREAD

**Discussion by smiririhem** [keycloak/keycloak#28853] [Q&A]
Hello community,

I'm currently working on Keycloak version 23.0.6. I'm trying to modify the attributes entered in the registration form, such as phone number and zip code. However, I encountered the following error message:

"Bad Request: Unrecognized field 'phoneNumber' (class org.keycloak.representations.account.UserRepresentation), not marked as ignorable (8 known properties: 'lastName', 'emailVerified', 'username', 'attributes', 'id', 'email', 'userProfileMetadata', 'firstName') at [Source: (org.jboss.resteasy.reactive.server.vertx.VertxInputStream); line: 1, column: 744] (through reference chain: org.keycloak.representations.account.UserRepresentation['phoneNumber'])"

Here's a snippet of the modifications I made in the account theme (`accountpage.js`):

errors: {
  username: '',
  firstName: '',
  lastName: '',
  phoneNumber:'',
  zipCode: ''
},
formFields: {
  username: '',
  firstName: '',
  lastName: '',
  email: '',
  attributes: {
    phoneNumber:'',
    zipCode: ''
  }
}
And in the handleSubmit function:

handleSubmit = event => {
  event.preventDefault();
  const form = event.target;
  const isValid = form.checkValidity();
  if (isValid) {
    const reqData = {
      ...this.state.formFields,
      attributes: {
        ...this.state.formFields.attributes,
        phoneNumber: this.state.formFields.attributes.phoneNumber,
        zipCode: this.state.formFields.attributes.zipCode
      }
    };
Additionally, i tried to add input for phonenumber

React.createElement(FormGroup, {
  label: Msg.localize("phoneNumber"),
  fieldId: "phoneNumber",
  helperTextInvalid: this.state.errors.phoneNumber,
  validated: this.state.errors.phoneNumber !== "" ? ValidatedOptions.error : ValidatedOptions.default,
},
  /*#__PURE__*/React.createElement(TextInput, {
    isRequired: false,
    type: "number",
    id: "phoneNumber",
    name: "phoneNumber",
    maxLength: 254,
    value: fields.phoneNumber,
    onChange: this.handleChange,
    validated: this.state.errors.phoneNumber !== "" ? ValidatedOptions.error : ValidatedOptions.default
  }))
How can I resolve this error? Any guidance would be appreciated.

**Comment by ThoreKr** [comment]
Why are you not using the declarative-user-profile, which gives you all that from a simple json config?

It is preview in 23 but became stable with 24.
